core/src/main/scala/scalaz/Foldable.scala

```diff
@@ -325,6 +325,9 @@ trait Foldable[F[_]]  { self =>
         else seen
     }.reverse
 
+  def distinctBy[A, B: Equal](fa: F[A])(f: A => B): IList[A] =
+    distinctE(fa)(Equal.equalBy(f))
```

Comment:
@jbgi Ah, missed that. I don't know, this seems to double the number of allocations compared to @xuwei-k's version.